types/utils.d.ts

```diff
@@ -24,6 +27,10 @@ export type ObjectOrFunc<TResult, TArg = {}> = ((arg: TArg) => TResult) | TResul
 
 export type Props = ObjectOf<any>
 export type ReactChildren = React.ReactNodeArray | React.ReactNode
+
+export type ReactPropsStrict<T> = { [K in keyof T]: NullableIfUndefined<T[K]> }
+export type ReactProps<T> = Extendable<ReactPropsStrict<T>>
```

Comment:
There is a small difference, `ReactPropsStrict` cannot be extended it's very useful for `Portal`, `Ref` and other component that don't allow to spread props.